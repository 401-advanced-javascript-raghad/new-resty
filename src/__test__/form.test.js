
import React from 'react';
import { shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import Form from '../../src/form/Form';

describe('<Form/>', ()=> {
    it('is alive at application start', () =>{
        let app = shallow(<Form />);
        expect(app.find('main').exists()).toBeTruthy();
    });

    // it('changes state on click', ()=> {
    //     let method = 'post';
    //     let url = 'localhost:3000/categories'
    //     let app = mount(<Form />);
    //     let button = app.find('button');
    //     button.simulate('click');
    //     expect(app.state({method: method, url: url})).toBe({method: 'post', url: 'localhost:3000/categories'});
    //     expect(app.find('.result').text()).toContain('post localhost:3000/categories');
    // });

    it('renders correctly', ()=> {
        const tree = renderer.create(<Form />).toJSON();
        expect(tree).toMatchSnapshot();
    });


});