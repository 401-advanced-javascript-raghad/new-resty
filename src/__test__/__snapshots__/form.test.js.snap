// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders correctly 1`] = `
<main
  className="main-class"
>
  <div
    className="url"
  >
    <label>
      URL:
    </label>
    <input
      onChange={[Function]}
      type="text"
    />
    <button
      onClick={[Function]}
    >
      GO!
    </button>
    <br />
  </div>
  <div
    className="method"
  >
    <input
      id="get"
      name="method"
      onChange={[Function]}
      type="radio"
      value="get"
    />
    <label>
      GET
    </label>
    <input
      id="post"
      name="method"
      onChange={[Function]}
      type="radio"
      value="post"
    />
    <label>
      POST
    </label>
    <input
      id="put"
      name="method"
      onChange={[Function]}
      type="radio"
      value="put"
    />
    <label>
      PUT
    </label>
    <input
      id="delete"
      name="method"
      onChange={[Function]}
      type="radio"
      value="delete"
    />
    <label>
      DELETE
    </label>
  </div>
  <div
    className="result"
  >
    
     
    
  </div>
</main>
`;
